### PR TITLE
Dump states

### PR DESCRIPTION
**Pull request by sbeaver-netgate**
Incorporates changes to:
diag_dump_states.php
diag_dump_states_sources.php
diag_reset_state.php

### CODE REVIEW

File: usr/local/www/diag_dump_states.php [hunk 1]
```diff
@@ -139,42 +132,36 @@ function(index,row) { jQuery(row).fadeOut(1000); }
...
+<?php
+		if (isset($_POST['filter']) && (is_ipaddr($_POST['filter']) || is_subnet($_POST['filter'])))
+		   print('<input type="submit" class="btn btn-danger btn-sm formbtn" name="killfilter" value="Kill">');
+?>
+
+	&nbsp;
+	</div>
+
+	</form>
+	<br>
```
Comment (SanderVanLeeuwen) [inline usr/local/www/diag_dump_states.php]: Please remove this and other `<br>`'s which are only used for spacing. That's _not_ what this element is for.

Comment (sbeaver-netgate) [inline usr/local/www/diag_dump_states.php]: Done

> On Apr 22, 2015, at 4:40 PM, Sander van Leeuwen notifications@github.com wrote:
> 
> In usr/local/www/diag_dump_states.php https://github.com/SjonHortensius/pfsense/pull/97#discussion_r28912524:
> 
> > -   <?=gettext("&nbsp;Current total state count")?>: <?= $current_statecount ?>
> > -   <div class="pull-right">
> > -       <?=gettext("Filter expression:")?>
> > -       <input type="text" name="filter" class="formfld search" value="<?=htmlspecialchars($_POST['filter'])?>" size="30" />
> > -       <input type="submit" class="btn btn-primary btn-sm formbtn" value="<?=gettext("Filter")?>" />
> >   +
> >   +<?php
> > -       if (isset($_POST['filter']) && (is_ipaddr($_POST['filter']) || is_subnet($_POST['filter'])))
> > -          print('<input type="submit" class="btn btn-danger btn-sm formbtn" name="killfilter" value="Kill">');
> >   +?>
> >   +
> > -   &nbsp;
> > -   </div>
> >   +
> > -   </form>
> > -   <br>
> >   Please remove this and other <br>'s which are only used for spacing. That's not what this element is for.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/97/files#r28912524.

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com mailto:sbeaver@netgate.com | www.netgate.com http://www.netgate.com/

File: usr/local/www/diag_dump_states.php [hunk 2]
```diff
@@ -54,11 +54,11 @@
 	return;
 }
 
-if (isset($_POST['filter']) && isset($_POST['killfilter'])) {
```
Comment (SjonHortensius) [inline usr/local/www/diag_dump_states.php:57]: Could you revert this and rename the input back to the original 'filter'?

File: usr/local/www/diag_dump_states.php [hunk 3]
```diff
@@ -116,137 +112,128 @@ function(index,row) { jQuery(row).fadeOut(1000); }
...
+
+$section->addInput(new Form_Input(
+	'filtertext',
+	'Filter expression',
+	'text',
+	htmlspecialchars($_POST['filtertext']),
+	['placeholder' => '']
+));
+
+$section->addInput(new Form_StaticText(
```
Comment (SjonHortensius) [inline usr/local/www/diag_dump_states.php:140]: Why don't you move the Button to the Form-constructor? I'm curious

File: usr/local/www/diag_dump_states.php [hunk 4]
```diff
@@ -116,137 +112,128 @@ function(index,row) { jQuery(row).fadeOut(1000); }
...
+
+$form = new Form(false);
+
+$section = new Form_Section('State filter');
+
+$section->addInput(new Form_Input(
+	'filtertext',
+	'Filter expression',
+	'text',
+	htmlspecialchars($_POST['filtertext']),
```
Comment (SjonHortensius) [inline usr/local/www/diag_dump_states.php:136]: htmlspecialchars not needed here

File: usr/local/www/diag_dump_states.php [hunk 5]
```diff
@@ -116,137 +112,128 @@ function(index,row) { jQuery(row).fadeOut(1000); }
...
-	$iface = convert_real_interface_to_friendly_descr($iface);
-
-	/* break up info and extract $srcip and $dstip */
-	$ends = preg_split("/\<?-\>?/", $info);
-	$parts = explode(":", $ends[0]);
-	$srcip = trim($parts[0]);
-	$parts = explode(":", $ends[count($ends) - 1]);
-	$dstip = trim($parts[0]);
+<script>
+	document.getElementById("filter").className = "btn btn-default btn-sm";
```
Comment (SjonHortensius) [inline usr/local/www/diag_dump_states.php:159]: Why not do this in the php-code?

File: usr/local/www/diag_dump_states.php [hunk 6]
```diff
@@ -116,137 +112,128 @@ function(index,row) { jQuery(row).fadeOut(1000); }
...
+	document.getElementById("killfilter").className = "btn btn-danger btn-sm";
+</script>
 
+<table class="table table-striped table-hover table-compact" summary="results">
+	<thead>
+		<tr class="info">
+			<th class="col-md-1"><?=gettext("Int")?></th>
```
Comment (SjonHortensius) [inline usr/local/www/diag_dump_states.php:166]: I think these classes aren't required

File: usr/local/www/diag_dump_states.php [hunk 7]
```diff
@@ -116,137 +112,128 @@ function(index,row) { jQuery(row).fadeOut(1000); }
...
 }
 
 if ($row == 0): ?>
-	<tr>
-		<td class="list" colspan="4" align="center" valign="top">
-		<?= gettext("No states were found.") ?>
-		</td>
-	</tr>
+		<tr>
+			<td colspan="4" align="center">
```
Comment (SjonHortensius) [inline usr/local/www/diag_dump_states.php:219]: Please remove the align too. Maybe wrap the message in an alert instead?

File: usr/local/www/diag_dump_states_sources.php [hunk 8]
```diff
@@ -104,54 +102,41 @@ function(index,row) { jQuery(row).fadeOut(1000); }
...
+	$tab_array = array();
+	$tab_array[] = array(gettext("States"), false, "diag_dump_states.php");
+	$tab_array[] = array(gettext("Source Tracking"), true, "diag_dump_states_sources.php");
+	$tab_array[] = array(gettext("Reset States"), false, "diag_resetstate.php");
+	display_top_tabs($tab_array);
+?>
 
 <!-- Start of tab content -->
+	<br />
+	<form action="<?=$_SERVER['SCRIPT_NAME']?>" method="get">
```
Comment (SjonHortensius) [inline usr/local/www/diag_dump_states_sources.php:118]: Is the reason you don't use a proper form here, that you want to use GET?

File: usr/local/www/diag_dump_states_sources.php [hunk 9]
```diff
@@ -104,54 +102,41 @@ function(index,row) { jQuery(row).fadeOut(1000); }
...
+		<div class="pull-right">
+			<?=gettext("Filter expression:")?>
+			<input type="text" name="filter" class="formfld search" value="<?=htmlspecialchars($_GET['filter'])?>" size="30" />
+			<input type="submit" class="btn btn-primary btn-sm formbtn" value="<?=gettext("Filter")?>" />
+			&nbsp;
```
Comment (SjonHortensius) [inline usr/local/www/diag_dump_states_sources.php:123]: not required

File: usr/local/www/diag_dump_states_sources.php [hunk 10]
```diff
@@ -104,54 +102,41 @@ function(index,row) { jQuery(row).fadeOut(1000); }
...
+		</div>
+	</form>
+	<br>
+	<div class="responsive">
+	<table class="table table-striped table-hover table-compact">
+		<thead>
+			<tr>
+				<th class="listhdrr" width="40%"><?=gettext("Source -> Destination")?></th>
```
Comment (SjonHortensius) [inline usr/local/www/diag_dump_states_sources.php:131]: all these classes are not required (the widths neither)

File: usr/local/www/diag_dump_states_sources.php [hunk 11]
```diff
@@ -169,44 +154,37 @@ function(index,row) { jQuery(row).fadeOut(1000); }
...
-				<img src='/themes/<?php echo $g['theme']; ?>/images/icons/icon_x.gif' height='17' width='17' border='0'
-					onclick="removeSource(<?php echo "'{$srcip}', '{$dstip}'"; ?>);" style='cursor:pointer;'
-					name='i:<?php echo "{$srcip}:{$dstip}"; ?>'
-					title='<?php echo gettext("Remove all source tracking entries from") . " {$srcip} " . gettext("to") . " {$dstip}";?>' alt='' />
-				</td>
-			  </tr>
-		<?php
+?>
+        	<tr name='r:<?="{$srcip}:{$dstip}" ?>'>
+        		<td class='listlr'><?=$info?></td>
```
Comment (SjonHortensius) [inline usr/local/www/diag_dump_states_sources.php:159]: These classes are not needed. Pretty sure the tr.name isn't either

File: usr/local/www/diag_dump_states_sources.php [hunk 12]
```diff
@@ -169,44 +154,37 @@ function(index,row) { jQuery(row).fadeOut(1000); }
...
 }
 else {
-	echo "<tr>
-			<td class='list' colspan='5' align='center' valign='top'>
-			  " . gettext("No source tracking entries were found.") . "
-			</td>
-		  </tr>";
+?>
+	        <tr>
+        		<td class="list" colspan="4" align="center" valign="top">
```
Comment (SjonHortensius) [inline usr/local/www/diag_dump_states_sources.php:176]: please remove class, align & valign here

File: usr/local/www/diag_dump_states_sources.php (hunk 10 above)
Comment (sbeaver-netgate) [inline usr/local/www/diag_dump_states_sources.php:131]: Sjon,

This looks like an old version.

Standby while I check please. There is no way I left all that junk in these files.

Steve

> On Apr 28, 2015, at 2:44 PM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/diag_dump_states_sources.php https://github.com/SjonHortensius/pfsense/pull/97#discussion_r29274787:
> 
> >  
> > -   <br />
> > -   <form action="<?=$_SERVER['SCRIPT_NAME']?>" method="get">
> > -       <div class="pull-right">
> > -           <?=gettext("Filter expression:")?>
> > -           <input type="text" name="filter" class="formfld search" value="<?=htmlspecialchars($_GET['filter'])?>" size="30" />
> > -           <input type="submit" class="btn btn-primary btn-sm formbtn" value="<?=gettext("Filter")?>" />
> > -           &nbsp;
> > -       </div>
> > -   </form>
> > -   <br>
> > -   <div class="responsive">
> > -   <table class="table table-striped table-hover table-compact">
> > -       <thead>
> > -           <tr>
> > -               <th class="listhdrr" width="40%"><?=gettext("Source -> Destination")?></th>
> >   all these classes are not required (the widths neither)
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/97/files#r29274787.

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com mailto:sbeaver@netgate.com | www.netgate.com http://www.netgate.com/

File: usr/local/www/diag_resetstate.php [hunk 13]
```diff
@@ -65,73 +75,65 @@
...
+if (isset($config['system']['lb_use_sticky']))
+	$tab_array[] = array(gettext("Source Tracking"), false, "diag_dump_states_sources.php");
+	$tab_array[] = array(gettext("Reset States"), true, "diag_resetstate.php");
+	display_top_tabs($tab_array);
+
+require('classes/Form.class.php');
+
+$form = new Form(new Form_Button(
+	'Submit',
+	gettext('Reset')
```
Comment (SjonHortensius) [inline usr/local/www/diag_resetstate.php:110]: gettext not needed in calls to Form_*

File: usr/local/www/diag_dump_states_sources.php (hunk 10 above)
Comment (SjonHortensius) [inline usr/local/www/diag_dump_states_sources.php:131]: I hope, I was pretty surpised as well. I'll have another look at the diff

**Comment by sbeaver-netgate** [comment]
PR and feature branched closed. A local restore from backup caused these files to become out of date.
Three new PRs will be submitted for each of the files

**Pull request closed without merging**